### PR TITLE
Feature/dev 6200

### PR DESCRIPTION
**Pull request by slemasne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/CONTRIBUTING.md)
- [ ] Tests pass

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import pandas as pd
+from flatten_json import flatten
+
+
+def lusid_respose_to_pd(lusid_response):
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: typo in "`repose`"

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    """
+    This function takes a LUSID API response and attempts to convert the response into a Pandas DataFrame or Series.
+    Not all LUSID API responses are the same datatype.
+    Therefore we need to implement some if/else conditional logic to see how the response should be converted.
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: This comment is more an implementation detail rather than user documentation, I would move to inside the function

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import pandas as pd
+from flatten_json import flatten
+
+
+def lusid_respose_to_pd(lusid_response):
+
+    """
+    This function takes a LUSID API response and attempts to convert the response into a Pandas DataFrame or Series.
+    Not all LUSID API responses are the same datatype.
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: List out the types of supported responses e.g. object containing `values`, dictionary etc.

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    elif hasattr(lusid_response, "to_dict"):
+
+        # The API responses for single LUSID objects have a "to_dict" attribute
+        # Example "Get Instrument" returns a dictionary with some key-value pairs for one instrument
+
+        pandas_object = pd.Series(flatten(lusid_response.to_dict(), "."))
+
+    else:
+
+        raise TypeError(
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: I wonder if rather than throwing an exception we convert the whole response raw? If you are using `lusidtools` for general use (i.e. not necessarily using `load_from_data_frame`) you might want to get the response and pre-process in a DataFrame in which case this would be a useful function 

@MikeMcGarry @thomasdotred what do you think?

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 1 above)
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: should this be `lusid_respose_to_data_frame` rather than `pd`? (adopting whatever convention we've already used for DataFrame in method names)

Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: I think we need to create the `__init__.py` and export the function

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
+from pathlib import Path
+import lusid
+import lusid.models as models
+import unittest
+from lusidtools.pandas_utils.lusid_pandas import lusid_respose_to_pd
+import pandas as pd
+
+test_instrument_response = lusid.models.instrument.Instrument(
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: move these into the test class definition

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+        instrument_series = lusid_respose_to_pd(test_instrument_response)
+        self.assertIsInstance(instrument_series, pd.Series)
+
+    def test_transaction_alias_response_to_series(self):
+
+        transaction_alias_series = lusid_respose_to_pd(transaction_alias_response)
+        self.assertIsInstance(transaction_alias_series, pd.Series)
+
+    def test_response_to_pd_fail(self):
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: also add a test for passing in `None` (for completeness)

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 1 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Thanks. That's corrected.

Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Sure, that's been added in latest commit.

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 4 above)
Comment (MikeMcGarry) [reply lusidtools/pandas_utils/lusid_pandas.py]: What do you mean by convert raw? The only exception here is if there is no to_dict method right? I think every model should have this method.

Also @slemasne I think this should always return a Pandas DataFrame. The Series and DataFrame have different attributes and methods (interfaces) so with this function you will never be sure what you will get back.

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 6 above)
Comment (MikeMcGarry) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Can we please also check some of the values on the output as well?

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 2 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Thanks. Corrected in latest commit.

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 4 above)
Comment (rizwansaeed) [reply lusidtools/pandas_utils/lusid_pandas.py]: sorry I meant to convert any arbitrary object to a DataFrame

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 3 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Thanks. Corrected in latest commit.

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 4 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Thanks. Thats updated.

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 1 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Agreed. Based on comments above, I wont return a pd.Series. Therefore **lusid_response_to_data_frame** is better name.

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 5 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Thats done.

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 6 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: That's done.
New check for 
(1) None
(2) Some df values
